### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by jrhea**
Basic skeleton in place with encoding and decoding logic.  Bluetooth transport is stubbed.

Tasks:
- [x] project config
- [x] code port
- [x] tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    then
+      echo "Only docs were updated, not running the CI."
+      exit
+    fi
+  - ./gradlew --no-daemon build
+
+script: ./gradlew --no-daemon test
+
+notifications:
+  email: false
```
Comment (decanus) [inline .travis.yml]: add new line at the end of file

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+os: osx
+language: kotlin
+osx_image: xcode10.1
```
Comment (decanus) [inline .travis.yml]: does not need to be osx for kotlin. Check:
 - https://github.com/android10/Android-CleanArchitecture-Kotlin/blob/master/.travis.yml
 - https://docs.travis-ci.com/user/languages/java/
 - https://github.com/square/leakcanary/blob/master/.travis.yml

or other kotlin projects.

File: ub/src/main/kotlin/com/e/ub/Peer.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    /// Initializes a peer with a specified id and list of known services.
+    ///
+    /// - Parameters:
+    ///     - id: The peer id.
+    ///     - services: The services a peer can knows.
+    constructor(id: Addr, services: List<UBID>) {
+        this.id = id
+        this.services = services
+    }
+}
```
Comment (decanus) [inline ub/src/main/kotlin/com/e/ub/Peer.kt]: add eol

File: ub/src/main/kotlin/com/e/ub/transports/BluetoothTransport.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    ///  The peers a specific transport can send messages to.
+    override val peers: List<Peer>
+        get() {
+            TODO()
+        }
+
+    override fun send(message: ByteBuffer, to: Addr){}
+    /// Listen implements a function to receive messages being sent to a node.
+    override fun listen(){}
+}
```
Comment (decanus) [inline ub/src/main/kotlin/com/e/ub/transports/BluetoothTransport.kt]: add eol

**Review by decanus — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (jrhea) [reply .travis.yml]: A couple of things.  I am not assuming this project is android only.  I can certainly change the os, but I am not sure what the point is.

**Review by decanus — APPROVED** [review]

**Pull request merged**